### PR TITLE
Basic pre-commit configuration

### PR DESCRIPTION
**Pull request by jpodivin**
#343 

Pre-commit hooks are now included to check for common style violations. These checks include remains of merge conflict delimiters, trailing whitespaces, pep8 violations and clang-format.

### CODE REVIEW

**Comment by inknos** [comment]
hey, thanks for the PR. I am using pre-commit for a while now and in the .pre-commit file I test rpmlint on `dnf5.spec`. could you add it? https://github.com/inknos/dnf5/blob/pre-commit/.pre-commit-config.yaml#L17

**Comment by gotmax23** [comment]
It's probably a good idea to add a CI job to run the pre-commit hooks as well.

**Comment by jpodivin** [comment]
@inknos I'm having trouble testing this locally.
When I try to run a build I'm getting multiple:

`collect2: error: ld returned 1 exit status`

File: test/libdnf/iniparser/test_iniparser.cpp [hunk 1]
```diff
@@ -127,15 +127,15 @@ void IniparserTest::test_iniparser2() {
 # Test comment2
 [section1]
 key1 = value1
-key2 = two line  
+key2 = two line
```
Comment (jpodivin) [inline test/libdnf/iniparser/test_iniparser.cpp:130]: Regarding this test. It's throwing errors for obvious reasons. I can either normalize the string to not trigger linters but then the assertions will have to be changed as well.

**Comment by inknos** [comment]
>  I'm having trouble testing this locally.
Testing now

> Edit: However, RPM lint is not released as pre-commit hook yet. Is it? So we would first need take care of that. Otherwise it would have to be installed manually, and I don't think that would be the most resilient setup.

It is not released and yes, it needs to be installed manually now. I am not sure on what could be the best way to add it as a requirement

File: test/libdnf/iniparser/test_iniparser.cpp (hunk 1 above)
Comment (inknos) [reply test/libdnf/iniparser/test_iniparser.cpp:130]: I would exclude the tests that involve text parsing directly by instructing pre-commit to not perform the checks on such files

Comment (jpodivin) [reply test/libdnf/iniparser/test_iniparser.cpp:130]: ok. That should let the checks pass.

Comment (inknos) [reply test/libdnf/iniparser/test_iniparser.cpp]: all is passing locally now.waiting for the build and tests to be green in GH

**Comment by inknos** [comment]
I would merge it and open an issue to include rpmlint in the pre-commit file, another issue to replace clang-format CI check with pre-commit. Thank you!

**Pull request merged**